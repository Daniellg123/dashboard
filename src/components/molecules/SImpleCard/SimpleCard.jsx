import PropTypes from "prop-types";
import { Card, Typography } from "@mui/material";
import ImageCard from "../../atoms/ImageCard/ImageCard";

import './styles.css';

/**
 * Componente para mostrar solo la imagen y el titulo
 */

function SimpleCard({ title, img }) {
    return (
        <Card className="simpleCard">
            <ImageCard imgDir={img} size={32} />
            <Typography variant="body1">{title}</Typography>
        </Card>
    );
};

SimpleCard.propTypes = {
    /**
     * Titulo del card
     */
    title: PropTypes.string,
    /**
     * Ruta de la imagen
     */
    img: PropTypes.string
};

export default SimpleCard;