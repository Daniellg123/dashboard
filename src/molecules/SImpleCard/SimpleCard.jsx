import { Card, CardMedia, CardContent, Typography } from "@mui/material";
import './styles.css';;

function SimpleCard({ title, img }) {
    return (
        <Card className="card">
            <CardMedia 
                component="img"
                sx={{ width: 32 }}
                image={img}
                alt={img}
            />
            <Typography variant="body1">{title}</Typography>
        </Card>
    );
};

export default SimpleCard;