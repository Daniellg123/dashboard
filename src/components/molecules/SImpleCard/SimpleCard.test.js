import { render } from '@testing-library/react';
import SimpleCard from './SimpleCard';

const props = {
    img: './img/logo1.png',
    title: 'Titulo',
};

it('renders correctrly', () => {
    const component = render(<SimpleCard {...props} />)
    expect(component).toMatchSnapshot();
});

it('Title is equal to title props', () => {
    const component = render(<SimpleCard {...props} title={'Tutiloo22'} />)
    component.getByText('Tutiloo22');
});