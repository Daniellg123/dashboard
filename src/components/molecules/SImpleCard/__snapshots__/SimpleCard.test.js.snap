// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctrly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root simpleCard css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <img
          alt="./img/logo1.png"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img css-1z7i81-MuiCardMedia-root"
          src="./img/logo1.png"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Titulo
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root simpleCard css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="./img/logo1.png"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img css-1z7i81-MuiCardMedia-root"
        src="./img/logo1.png"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Titulo
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
